### ISSUE TITLE
are the `mentions_mentioner_idx` and `mentions_mentionee_idx` indexes needed?

### ISSUE THREAD

**Issue by clairity** [pmviva/mention_system#3]
i am going through my codebase and cleaning up extraneous indexes and wondered if the `mentions_mentioner_idx` and `mentions_mentionee_mentioner_idx` indexes are needed? i ask because the migration already creates two indexes

  1. index_mentions_on_mentionee_type_and_mentionee_id
  2. index_mentions_on_mentioner_type_and_mentioner_id

as defined implicitly by these polymorphic references:

```ruby
class CreateMentions < ActiveRecord::Migration[5.0]
  def change
    create_table :mentions do |t|
      t.references  :mentionee, polymorphic: true
      t.references  :mentioner, polymorphic: true
      ...
    end
     
    ...

  end
end
```

so are these two additional indexes required?

```ruby
class CreateMentions < ActiveRecord::Migration[5.0]
  def change
    create_table :mentions do |t|
      ...
    end

    add_index :mentions, [:mentionee_id, :mentionee_type], name: "mentions_mentionee_idx"
    add_index :mentions, [:mentioner_id, :mentioner_type], name: "mentions_mentioner_idx"
    ...
  end
end
```

the main difference is the order of the index elements:

  1. [:mentionee_type, :mentionee_id] rather than [:mentionee_id, :mentionee_type]
  2. [:mentioner_type, :mentioner_id] rather than [:mentioner_id, :mentioner_type]

if these two defined indexes are indeed required, then are the implicitly created ones needed? not a huge deal, but i'd thought i'd ask since i couldn't find an answer elsewhere. thanks!

**Comment by pmviva** [comment]
Probably it has to do with the fact that in rails 4.1 when I first started developing this gem, adding an index to an association in a migration was optional.

In rails migration guide for 4.1 when discussing t.references it uses index: true for forcing index creation.
It was not the default behavior I guess, I included the add_index manually to be more verbose and explicit about index creation.

Those indexes are not needed since rails 5 creates indexes by default on references.
